### PR TITLE
Allow splitBDFAt functions to work with filepaths containing `.` and relative file paths

### PR DESCRIPTION
**Pull request by codles**

### CODE REVIEW

File: src/BDF.jl [hunk 1]
```diff
@@ -737,7 +737,7 @@ function splitBDFAtTrigger(fname::AbstractString, trigger::Integer; from::Real=0
     stopPoints =  [sepPoints; size(data)[2]] 
     
     for i=1:nChunks
-        thisFname = string(split(fname, ".")[1], "_", i, ".", split(fname, ".")[2])
+        thisFname = joinpath(dirname(fname), basename(fname)[1:end-4] * "_" * string(i) * basename(fname)[end-3:end])
```
Comment (codles) [inline src/BDF.jl]: Could probably have just done 
```julia
joinpath(fname[1:end-4] * "_" * string(i) * basename(fname)[end-3:end])
```
I was hoping there was an `extension(fname)` function then it could have been 

```julia
joinpath(dirname(fname), basename(fname)[1:end-4] * "_" * string(i) * extension(fname))
```

**Comment by sam81** [comment]
Thanks, looks good. Maybe we could refine it to cover corner cases, e.g. somebody saving the BDF file without any extension at all, but I think that should be rare.

It seems that the Travis CI build failed for the changes in the test directory files, so we may have to revert those. I'm merging for now.

**Comment by codles** [comment]
I committed the test change first to highlight the problem, then committed the fix. Sorry that's a tad confusing, I'll just do one complete request next time.

Regarding the corner cases, I agree it should be rare, but ideally something to check. This works for now.

Thanks for accepting the PR.

**Comment by sam81** [comment]
OK, all good then! I'll tag a new release with the fixes as soon as I have some time. Thanks.
